### PR TITLE
Feature/admin can import rewards from csv

### PR DESCRIPTION
**Pull request by Gludek**
Allow Admin to import rewards from CSV file

In this PR I created service object and controller actions to allow importing rewards from CSV file.
To allow possibility of columns not being in order they exist in model I'm creating hash where hash fields are created from first row of imported file.

### CODE REVIEW

File: app/services/application_service.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class ApplicationService
+  attr_reader :result
+
+  def initialize
+    raise 'Implement initilization'
```
Comment (ciejj) [inline app/services/application_service.rb:5]: I would suggest getting rid of that

File: app/services/application_service.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class ApplicationService
```
Comment (ciejj) [inline app/services/application_service.rb:1]: inheritance is rather considered as antipattern nowadays,

File: app/services/reward_importer.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+class RewardImporter < ApplicationService
+  attr_reader :rewards
+
+  # rubocop:disable Lint/MissingSuper
+  def initialize(rewards_file)
+    @result = { success?: nil, failure: Hash.new(error: '', row: nil) }
```
Comment (ciejj) [inline app/services/reward_importer.rb:6]: this is way too much logic in the initialize - initialize should only do what it says - initialize instance variables -> all logic should be moved to call method

File: app/services/application_service.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  def initialize
+    raise 'Implement initilization'
+  end
+
+  def self.call(...)
+    new(...).call
+  end
+
+  def success?
```
Comment (ciejj) [inline app/services/application_service.rb:12]: why not simply return `true` or `false` as a result?

File: app/views/admins/rewards/import_form.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    </label>
+  </div>
+
+  <div class="actions container">
+    <%= f.submit 'Import Rewards', class:'button is-primary' %>
+  </div>
+<% end %>
+<div class='columns is-vcentered mt-1 has-text-centered'>
+  <%= link_to 'Back', admins_rewards_path, class:'column' %>
+</div>
```
Comment (ciejj) [inline app/views/admins/rewards/import_form.erb:22]: missing newline

File: config/routes.rb [hunk 6]
```diff
@@ -15,7 +15,10 @@
       get :add_kudos, on: :collection, to: "employees#add_kudos_form"
       patch :add_kudos, on: :collection
     end
-    resources :rewards
+    resources :rewards do
+      get :import, on: :collection, to: "rewards#import_form"
+      put :import, on: :collection
```
Comment (ciejj) [inline config/routes.rb:20]: I think it should be `post`

**Review by ciejj — APPROVED** [review]
Well done, but in my opinion, the service is a bit overcomplicated - why do you write your own uniqueness validation instead of using one provided already by Active Record?

File: app/services/application_service.rb (hunk 4 above)
Comment (Gludek) [reply app/services/application_service.rb:12]: I also need to pass information about error if it occurs, that's why I'm creating hash that has information about success and information about error if needed.

**Pull request merged**